proxy/apifilterproxy/util/util.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/base64"
+	log "github.com/Sirupsen/logrus"
+	"github.com/pborman/uuid"
+)
+
+func SignString(stringToSign []byte, sharedSecret []byte) string {
+	h := hmac.New(sha512.New, sharedSecret)
+	h.Write(stringToSign)
+
+	signature := h.Sum(nil)
+	encodedSignature := base64.URLEncoding.EncodeToString(signature)
+
+	log.Debugf("Signature generated: %v", encodedSignature)
+
+	return encodedSignature
+}
+
+func GenerateUUID() string {
```

Comment:
we have to document on how we generate the UUID so the people writing generic http proxy implementation know how to parse it